docs/dev-guide/contrib-guides-taskfiles.md

```diff
@@ -0,0 +1,161 @@
+# Taskfiles
+
+Follow the guidelines below when writing and updating [Taskfiles]. Note that neither the guidelines
+nor examples are written as a rigorous specification, but they should be sufficient to understand
+the spirit of each guideline.
+
+## Variable naming
+
+1. Variables should be named using `SCREAMING_SNAKE_CASE`. 
+2. Variables in the global scope (as opposed to a task's scope) should be prefixed with `G_` to
+   avoid conflicts with local variables.
+
+**Example**
+
+```yaml
+vars:
+  G_BUILD_DIR: "build"
+
+binaries:
+  vars:
+    BIN_BUILD_DIR: "{{.G_BUILD_DIR}}/bin"
+```
+
+## Paths in commands
+
+All paths used in commands should be quoted.
+
+**Example**
+
+```yaml
+my-task:
+  cmds:
+    - "touch '{{.FILE_PATH}}'"
+    - |-
+      echo "Hello, world" > "{{.FILE_PATH}}"
+```
+
+## Built-in variables
+
+1. Don't use the variable `{{.ROOT_TASKFILE}}`
+   * v3.35.1 has a bug that makes it equivalent to `ROOT_DIR` (i.e., the parent directory of the root
+     Taskfile, rather than the path to the root Taskfile).
+2. When using the variable `{{.TASK}}` in a task within a namespace (e.g., it's in a nested
+   Taskfile), replace the `:` in the name with `#` as follows: `{{.TASK | replace \":\" \"#\"}}`.
+   * This ensures the task name can be used as a filename.
+
+## `sources` and `generates`
```

Comment:
Do we want to mention how (and why) we want to checksum certain tasks instead of this `sources` + `generates` pattern?
If we're short of time, we can do that in another PR.

Reply:
Good point. Added.